### ISSUE TITLE
Mysql doesn't start

### ISSUE THREAD

**Issue by corzel** [littlebizzy/slickstack#49]
Is not possible start mysql sever.

```
$ sudo systemctl start mysql
Job for mysql.service failed because the control process exited with error code.
See "systemctl status mysql.service" and "journalctl -xe" for details.
$ systemctl status mysql.service
● mysql.service - MySQL Community Server
     Loaded: loaded (/lib/systemd/system/mysql.service; enabled; vendor preset: enabled)
     Active: activating (start) since Mon 2020-07-13 20:40:03 UTC; 1min 29s ago
    Process: 5916 ExecStartPre=/usr/share/mysql/mysql-systemd-start pre (code=exited, status=0/SUCCESS)
   Main PID: 5924 (mysqld)
     Status: "Server startup in progress"
      Tasks: 16 (limit: 4683)
     Memory: 391.8M
     CGroup: /system.slice/mysql.service
             └─5924 /usr/sbin/mysqld
$ journalctl -xe
Hint: You are currently not seeing messages from other users and the system.
      Users in groups 'adm', 'systemd-journal' can see all messages.
      Pass -q to turn off this notice.
-- A start job for unit UNIT has finished successfully.
--
-- The job identifier is 2.
Jul 13 19:12:52 server-two systemd[944]: Reached target Main User Target.
-- Subject: A start job for unit UNIT has finished successfully
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- A start job for unit UNIT has finished successfully.
--
-- The job identifier is 1.
Jul 13 19:12:52 server-two systemd[944]: Startup finished in 75ms.
-- Subject: User manager start-up is now complete
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- The user manager instance for user 1001 has been started. All services queued
-- for starting have been started. Note that other services might still be starting
-- up or be started at any later time.
--
-- Startup of the manager took 75266 microseconds.
Jul 13 19:21:07 server-two sudo[1377]: pam_unix(sudo:auth): Couldn't open /etc/securetty: No such file or directory

```

**Comment by damiafaw** [comment]
same for me also, mysql does not load/start and therefore database connection.
Doesnt seem to have created the database either and I have run through it many many times now.

**Comment by damiafaw** [comment]
Tried again this morning, after it completed - there was no database created and no user.
Created database, created the user defined in the config file (that will be used in wp-config.php) and did the permissions;
mysql> CREATE USER 'xxxxxxx'@'%' IDENTIFIED BY 'password';
mysql> GRANT ALL PRIVILEGES ON *.* TO 'xxxxxxxxx'@'%' WITH GRANT OPTION;
mysql> FLUSH PRIVILEGES;

Now it works - so something is broken in the SQL creation process

**Comment by jessuppi** [comment]
This is a known issue as we comb through updating the MySQL installation process to better support MySQL 8.0.x. We've begun splitting the `ss-install` into various sub-scripts for better organization, including `ss-install-mysql`:

https://github.com/littlebizzy/slickstack/blob/master/ss-install-mysql.txt

We are moving toward using the `auth_socket` for `root` user and initial setup process. A new user `admin` will be created for all database management (and PHP tools, etc) as needed.

The challenge is running setup using this method but then using TCP (127.0.0.1) for the WordPress database.

All feedback is welcome, there is also a brief discussion in our Facebook group too. Hopefully within the next few days this gets resolved, it was also complicated by some changes made in MySQL 8.0.17/18 as well...

**Comment by damiafaw** [comment]
Thank you for the information - I kept seeing changes to the mysql install file and got excited and retried it a few times last few days.
Just running some tests against a copy of my live site at the moment, so was wanting to install from new and test - but the SQL was failing on install.

Keep up the good work, love how simple SlickStack is.

**Comment by jessuppi** [comment]
Thanks again for all the feedback on this thread.

We did successfully simplify the MySQL installation although still cleaning up the code a bit. This was complicated due to breaking changes that MySQL introduced in minor version updates (definitely not _semantic_ versioning!).

But it's been stable for more than a year now, which is good news.

- we also introduced a new script `ss-install-mysql-database` to separate that functionality
- Unix sockets will be default going forward for `root` MySQL user
- TCP will be default for non-root MySQL user (e.g. WordPress database user) to support remote databases, etc.

Also keep an eye on `ss-functions` as we optimize the bash aliases for `mysql` commands... all feedback is welcome, as we need to ensure these commands work on various Ubuntu LTS / MySQL versions.

Ref: https://github.com/littlebizzy/slickstack/blob/master/bash/ss-functions.txt
